tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/TspTool/TspTool.cs

```diff
@@ -0,0 +1,230 @@
+                    await HandleConvertCommand(ctx, ct);
+                    return 0;
+                default:
+                    logger.LogError($"Unknown command: {command}");
+                    return 1;
+            }
+        }
+
+        private Task HandleConvertCommand(InvocationContext ctx, CancellationToken ct)
+        {
+            var swaggerReadme = ctx.ParseResult.GetValueForOption(swaggerReadmeArg);
+            var outputDirectory = ctx.ParseResult.GetValueForOption(outputDirectoryArg);
+            var isArm = ctx.ParseResult.GetValueForOption(isArmOption);
+            var fullyCompatible = ctx.ParseResult.GetValueForOption(fullyCompatibleOption);
+
+            TspToolResponse result = ConvertSwagger(swaggerReadme, outputDirectory, isArm, fullyCompatible);
+            ctx.ExitCode = ExitCode;
+            output.Output(result.ToString());
+            return Task.CompletedTask;
+        }
+
+        [McpServerTool(Name = "ConvertSwaggerToTypeSpec"), Description(@"**Call this tool when trying to convert an existing Azure service to TypeSpec.**
+        This command should only be ran once to get started working on a TypeSpec project.
+        Verify whether the source swagger describes an Azure Resource Management (ARM) API or a data plane API if unsure.
+        Pass in the `pathToSwaggerReadme` which is the path to the swagger README file.
+        Pass in the `outputDirectory` where the TypeSpec project should be created. This must be an existing empty directory.
+        Pass in `isAzureResourceManagement` to indicate whether the swagger is for an Azure Resource Management (ARM) API.
+        This should be true if the swagger's path contains `resource-manager`.
+        Pass in `fullyCompatible` to indicate whether the generated TypeSpec project should be fully compatible with the swagger.
+        It is recommended not to set this to `true` so that the converted TypeSpec project
+        leverages TypeSpec built-in libraries with standard patterns and templates.
+        Returns path to the created project.")]
+        public TspToolResponse ConvertSwagger(string pathToSwaggerReadme, string outputDirectory, bool? isAzureResourceManagement = null, bool? fullyCompatible = null)
+        {
+            try
+            {
+                logger.LogInformation("Converting swagger to TypeSpec: {pathToSwaggerReadme}, output: {outputDirectory}, isArm: {isArm}, fullyCompatible: {fullyCompatible}",
+                    pathToSwaggerReadme, outputDirectory, isAzureResourceManagement, fullyCompatible);
+
+                // Validate pathToSwaggerReadme
+                string? readmeValidationResult = ValidateSwaggerReadme(pathToSwaggerReadme);
+                if (readmeValidationResult != null)
+                {
+                    SetFailure();
+                    return new TspToolResponse
+                    {
+                        ErrorMessage = readmeValidationResult
```

Comment:
You can use the `ResponseError` or `ResponseErrors` property from the base `Response` class for this, if desired.